independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/InjectionTargetInfo.java

```diff
@@ -0,0 +1,27 @@
+package io.quarkus.arc.processor;
+
+public interface InjectionTargetInfo {
+
+    TargetKind kind();
+
+    default BeanInfo asBean() {
+        throw new IllegalStateException("Not a bean");
+    }
+
+    default ObserverInfo asObserver() {
+        throw new IllegalStateException("Not an observer");
+    }
+
+    default DisposerInfo asDisposer() {
+        throw new IllegalStateException("Not a disposer");
+    }
+
+    static enum TargetKind {
```

Comment:
Minor nitpack (feel free to ignore) -> `static` is redundant

Reply:
Ah, you're right of course. Let's get rid of the redundant static! ;-)